include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -0,0 +1,508 @@
+     * related code and should by enclosed in a try catch block.
+     *
+     * \throws seqan3::parser_interruption
+     *
+     * \details
+     *
+     * When no specific key words are supplied, the seqan3::argument_parser 
+     * starts to process the command line for specified options, flags and
+     * positional options.
+     *
+     * The parser behaves differently when the the given command line (argv) 
+     * contains the following keywords (in order of checking):
+     *
+     * - **--version** Prints the version information and throws 
+     *                 a seqan3::parser_interruption.
+     * - **--copyright** Prints the copyright information and throws 
+     *                   an seqan3::parser_interruption.
+     * - **-h/--help** Prints the help page and throws 
+     *                 a seqan3::parser_interruption.
+     * - **-hh/--advanced-help** Prints the help page including advanced options
+     *                           and throws a seqan3::parser_interruption.
+     * - **--export-help [format]** Prints the help page information in the 
+     *                              given format (html/man/ctd) and throws a
+     *                              seqan3::parser_interruption.
+     *
+     */
+    void parse()
+    {
+        if (argc_ == 1) // no arguments provided
+            print_short_help();
+        else if (std::find(argv_.begin(), argv_.end(), "--version") != argv_.end())
+            print_version(); // test for --version option here and not on construction so all meta information are compiled.
+        else if (std::find(argv_.begin(), argv_.end(), "--copyright") != argv_.end())
+            print_long_copyright();
+        else
+            std::visit([this] (auto & f) {f.parse(info);}, format);
+    }
+
+    //!\name Structuring the Help Page
+    //!\{
+
+    /*!\brief Adds an help page section to the seqan3::argument_parser.
+     *
+     * \details This only affects the help page or export of information.
+     */
+    void add_section(std::string const & title)
+    {
+        std::visit([&] (auto & f) {f.add_section(title);}, format);
+    }
+
+    /*!\brief Adds an help page subsection to the seqan3::argument_parser.
+     *
+     * \details This only affects the help page or export of information.
+     */
+    void add_subsection(std::string const & title)
+    {
+        std::visit([&] (auto & f) {f.add_subsection(title);}, format);
+    }
+
+    /*!\brief Adds an help page text line to the seqan3::argument_parser.
+     *
+     * \details This only affects the help page or export of information.
+     */
+    void add_line(std::string const & text, bool line_is_paragraph)
+    {
+        std::visit([&] (auto & f) {f.add_line(text, line_is_paragraph);}, format);
+    }
+
+    /*!\brief Adds an help page list item (key-value) to the seqan3::argument_parser.
+     *
+     * \details This only affects the help page or export of information.
```

Comment:
hm the option is called `--export-help [format]` so I thought to keep it similar.
maybe `This only affects the help page and other output formats` ?